reinforcement_learning/README.md

```diff
@@ -0,0 +1,93 @@
+
+  * Gymnasium environments
+  * Training scripts
+
+## Gymnasium environments
+
+To use the environments, you first need to register them with Gymnasium, to do so you need to execute their attached script.
+For example, to register the `DialogueEnvMovieBot-v0` environment, you need to execute the following command:
+
+```bash
+python -m reinforcement_learning.environment.dialogue_env_moviebot
+```
+
+Note that to use the environments you will need to import them from the `reinforcement_learning.environment` package, for example:
+
+```python
+from reinforcement_learning.environment.dialogue_env_moviebot import DialogueEnvMovieBot
+```
+
+## Training scripts
+
+Currently, there are two reinforcement learning algorithms supported:
+
+  * Deep Q-Learning
+  * Advantage Actor-Critic
+
+Each algorithm has its own trainer, which can be found in the `reinforcement_learning.training` package.
+
+Before training a dialogue policy, you need to prepare a configuration file.
+A default configuration for each supported algorithm is provided in the `config/rl` folder.
+
+A dialogue policy is trained by executing the following command:
+
+```bash
+python -m reinforcement_learning.train_dialogue_policy -c <training_configuration>
+```
+
+The training logs are stored in Weight and Biases.
+
+### Training Configuration
+
+The training configuration is used to specify paths to participants configuration files, model's parameters, training hyperparameters, and other options related to training. Below is an empty configuration file with main options:
+
+```yaml
+# Participants' configuration
+usersim_config: <path to user simulator configuration>
+agent_config: <path to IAI MovieBot configuration>
+
+# Model's parameters
+model:
+  algorithm: <dqn OR a2c>
+  policy_input_size: 
+  hidden_size: 
+  model_path: <path to saved model>
+
+# Training hyperparameters
+hyperparams:
+  hyperparam_name: hyperparam_value
+  ...
+
+use_intents: 
+turn_penalty: 
+no_nlu: 
+
+debug: False
+```
+
+**Participants' configuration**: The participants' configuration files are used to specify the parameters of the user simulator and the agent.
+
+**Model's parameters**: The model's parameters are used to specify the parameters of the dialogue policy model. The `algorithm` parameter is used to specify the type of the algorithm used, which can be either `dqn` or `a2c`. The `policy_input_size` parameter is used to specify the size of the input of the policy network. The `hidden_size` parameter is used to specify the size of the hidden layers of the policy network. The `model_path` parameter is used to specify the path to the saved model.
+
+**Hyperparameters**: This section is reserved for training hyperparameters such as learning_rate, discount factor, etc. The hyperparameters are algorithm-specific and can be loaded easily in a dictionary.
+
+**use_intents**: Specifies whether the markovian state should include intents.
```

Comment:
User, agent or both? Current or previous turn?

Reply:
I added details to the description, please check that it is clarified.